common/app/components/VideoPlayer.tsx

```diff
@@ -1025,6 +1025,41 @@ export default function VideoPlayer({
         );
     }, [keyBinder, playerChannel]);
 
+    useEffect(() => {
+        const unbindRepeat = keyBinder.bindToggleRepeat(
+            (event) => {
+                event.preventDefault();
+                if (playMode === PlayMode.repeat) {
+                    setPlayMode(PlayMode.normal);
+                } else {
+                    setPlayMode(PlayMode.repeat);
+                }
+            },
+            () => false
+        );
+
+        return () => unbindRepeat();
+    }, [keyBinder, playMode, setPlayMode]);
+
+    useEffect(() => {
+        if (playMode !== PlayMode.repeat || showSubtitlesRef.current.length === 0) {
```

Comment:
`repeat` mode currently will only work when a video is open. To maintain consistency with the other play modes I would implement this logic inside `Player.tsx` instead.